test/unit/projectTest.ts

```diff
@@ -406,6 +406,25 @@ describe('SfdxProject', () => {
       expect(actual).to.deep.equal(expected);
     });
 
+    it('should set the a single package entry as default', async () => {
```

Comment:
Add another test for one with no packages and you should see the index error I pointed out above.